gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/tpgm/aggregation/AverageDurationTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.tpgm.aggregation;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.gradoop.common.model.api.entities.EPGMEdgeFactory;
+import org.gradoop.common.model.api.entities.EPGMVertexFactory;
+import org.gradoop.common.model.impl.pojo.temporal.TemporalEdge;
+import org.gradoop.common.model.impl.pojo.temporal.TemporalElement;
+import org.gradoop.common.model.impl.pojo.temporal.TemporalVertex;
+import org.gradoop.common.model.impl.properties.Properties;
+import org.gradoop.flink.model.GradoopFlinkTestBase;
+import org.gradoop.flink.model.impl.tpgm.TemporalGraph;
+import org.gradoop.flink.model.impl.tpgm.TemporalGraphFactory;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static org.gradoop.common.model.impl.pojo.temporal.TemporalElement.DEFAULT_TIME_FROM;
+import static org.gradoop.common.model.impl.pojo.temporal.TemporalElement.DEFAULT_TIME_TO;
+import static org.gradoop.common.model.impl.properties.PropertyValue.create;
+import static org.gradoop.flink.model.api.tpgm.functions.TemporalAttribute.TRANSACTION_TIME;
+import static org.gradoop.flink.model.api.tpgm.functions.TemporalAttribute.VALID_TIME;
+import static org.gradoop.flink.model.impl.operators.aggregation.functions.average.Average.IGNORED_VALUE;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test for the {@link AverageDuration} aggregate function.
+ */
+public class AverageDurationTest extends GradoopFlinkTestBase {
+
+  /**
+   * Test the implementation of {@link AverageDuration#getIncrement(TemporalElement)} for
+   * transaction time.
+   */
+  @Test
+  public void testGetIncrementForTxTime() {
+    AverageDuration function = new AverageDuration("", TRANSACTION_TIME);
+    TemporalElement testElement = getConfig().getTemporalGraphFactory().getVertexFactory()
+      .createVertex();
+    testElement.setTransactionTime(Tuple2.of(DEFAULT_TIME_FROM, DEFAULT_TIME_TO));
```

Comment:
Using this without the default check, we will have a numeric overflow. So maybe use a current system time as default from value.